modular_skyrat/modules/deforest_medical_items/code/chemicals/twitch.dm

```diff
@@ -0,0 +1,213 @@
+	UnregisterSignal(our_guy, COMSIG_MOVABLE_HEAR)
+	if(overdosed)
+		UnregisterSignal(our_guy, COMSIG_ATOM_PRE_BULLET_ACT)
+
+	if(constant_dose_time < 60) // Anything less than this and you'll come out fiiiine, aside from a big hit of stamina damage
+		our_guy.visible_message(
+			span_danger("[our_guy] suddenly slows from their inhuman speeds, coming back with a wicked nosebleed!"),
+			span_danger("You suddenly slow back to normal, a stream of blood gushing from your nose!")
+		)
+		our_guy.adjustStaminaLoss(constant_dose_time)
+	else // Much longer than that however, and you're not gonna have a good day
+		our_guy.visible_message(
+			span_danger("[our_guy] suddenly snaps back from their inhumans speeds, coughing up a spray of blood!"),
+			span_danger("As you snap back to normal speed you cough up a worrying amount of blood. You feel like you've just been run over by a power loader.")
+		)
+		our_guy.spray_blood(our_guy.dir, 2) // The before mentioned coughing up blood
+		our_guy.emote("cough")
+		our_guy.adjustStaminaLoss(constant_dose_time)
+		our_guy.adjustOrganLoss(ORGAN_SLOT_HEART, 0.3 * constant_dose_time) // Basically you might die
+
+	if(!our_guy.hud_used)
+		return
+
+	var/atom/movable/plane_master_controller/game_plane_master_controller = our_guy.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
+
+	game_plane_master_controller.remove_filter(TWITCH_SCREEN_FILTER)
+	game_plane_master_controller.remove_filter(TWITCH_SCREEN_BLUR)
+
+
+/// Leaves an afterimage behind the mob when they move
+/datum/reagent/drug/twitch/proc/on_movement(mob/living/carbon/our_guy, atom/old_loc)
+	SIGNAL_HANDLER
+	new /obj/effect/temp_visual/decoy/twitch_afterimage(old_loc, our_guy)
+
+
+/// Tries to dodge incoming bullets if we aren't disabled for any reasons
+/datum/reagent/drug/twitch/proc/dodge_bullets(mob/living/carbon/human/source, obj/projectile/hitting_projectile, def_zone)
+	SIGNAL_HANDLER
+
+	if(HAS_TRAIT(source, TRAIT_INCAPACITATED))
+		return NONE
+	source.visible_message(
+		span_danger("[source] effortlessly dodges [hitting_projectile]!"),
+		span_userdanger("You effortlessly evade [hitting_projectile]!"),
+	)
+	playsound(source, pick('sound/weapons/bulletflyby.ogg', 'sound/weapons/bulletflyby2.ogg', 'sound/weapons/bulletflyby3.ogg'), 75, TRUE)
+	source.add_filter(TWITCH_BLUR_EFFECT, 2, gauss_blur_filter(5))
+	addtimer(CALLBACK(source, TYPE_PROC_REF(/datum, remove_filter), TWITCH_BLUR_EFFECT), 0.5 SECONDS)
+	return COMPONENT_BULLET_PIERCED
+
+
+/datum/reagent/drug/twitch/on_mob_life(mob/living/carbon/our_guy, seconds_per_tick, times_fired)
+	. = ..()
+
+	constant_dose_time += seconds_per_tick
+
+	our_guy.adjustOrganLoss(ORGAN_SLOT_HEART, 0.1 * REM * seconds_per_tick)
+
+	if(locate(/datum/reagent/drug/kronkaine) in our_guy.reagents.reagent_list) // Kronkaine, another heart-straining drug, could cause problems if mixed with this
+		our_guy.ForceContractDisease(new /datum/disease/adrenal_crisis(), FALSE, TRUE)
```

Comment:
If you already have the disease then itll ignore you, and if you're immune to it, itll skip over as well, otherwise its the same sort of behavior kronkaine itself has.

Reply:
Hmmm... Okay, I don't like it, but okay.